messages.go

```diff
@@ -0,0 +1,140 @@
+package dht
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/protocol"
+	"github.com/libp2p/go-libp2p-core/routing"
+	record "github.com/libp2p/go-libp2p-record"
+	"github.com/multiformats/go-multihash"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+
+	recpb "github.com/libp2p/go-libp2p-record/pb"
+
+	pb "github.com/libp2p/go-libp2p-kad-dht/pb"
+)
+
+type ProtocolMessenger struct {
```

Comment:
So, people who want to simply use this without running a DHT instance need to get peers to talk to(who also support the DHT protocol) out of band ? How do they discover such peers ?

Docs explaining why this was created/exists with a note on how to use it without starting a DHT instance would be great.

Reply:
Also, What do you think of calling this the `DhtRpcClient` ? I think it conveys the use better.

Reply:
> So, people who want to simply use this without running a DHT instance need to get peers to talk to(who also support the DHT protocol) out of band ?

Yep.

> Also, What do you think of calling this the DhtRpcClient ? I think it conveys the use better.

Sure, I was trying to avoid the RPC terminology since it seems a bit loaded, but doesn't bother me if you think it adds clarity.

Reply:
Moved this into it's own package currently called `wire`. I'm not really sold on describing this as an RPC mostly because I think of RPCs as something you execute between machines you control (e.g. client tells server to do something, or primary node instructs secondary nodes).

Could be convinced of a better naming schemes here (especially since its now in it's own package).